### DISCUSSION TITLE
关于增加文献名与文献标识符之间的空格

### DISCUSSION THREAD

**Discussion by earthmoon** [hushidong/biblatex-gb7714-2015#167] [Q&A]
您好，

文末文献，比如`文献名[J]`，可否设置一个选项是其可以为`文献名 [J]`？

谢谢

**Comment by hushidong** [comment]
你使用我刚刚传到github上的bbx/cbx，然后在导言区增加：

```
\DeclareDelimFormat[bib,biblist]{titletypedelim}{\space}
```

就能达到增加空格的目的

比如：

![图片](https://github.com/hushidong/biblatex-gb7714-2015/assets/20421795/ac1ccb41-7ae3-4fd6-a047-7da94dafb508)

**Reply by earthmoon** [reply]
我这里为什么失败了。
```
\documentclass{article}
\usepackage{ctex}

\begin{filecontents}[force]{\jobname.bib}
@article{mikkonen2018,
	title = {Geochemical Indices and Regression Tree Models for Estimation of Ambient Background Concentrations of Copper, Chromium, Nickel and Zinc in Soil},
	author = { family=Graaff, given=Robert, prefix=van de, useprefix=false and Clarke, Bradley O. and Dasika, Raghava and Wallis, Christian J. and Reichman, Suzie M.},
	date = {2018-11-01},
	journaltitle = {Chemosphere},
	volume = {210},
	pages = {193--203},
	issn = {0045-6535},
	doi = {10.1016/j.chemosphere.2018.06.138},
	}
@article{first,
	title = {The firstl},
	author = { Go},
	date = {6},
	journaltitle = {5},
	volume = {4},
	pages = {3},
	issn = {2},
	doi = {1},
	}
@article{second,
	title = {The firstl},
	author = { Rr},
	date = {6},
	journaltitle = {5},
	volume = {4},
	pages = {3},
	issn = {2},
	doi = {1},
	}
\end{filecontents}

\usepackage[style=gb7714-2015ay]{biblatex} %,defernumbers=true
\addbibresource{\jobname}

\DeclareDelimFormat[bib,biblist]{titletypedelim}{\space}
\begin{document}
\newbibmacro*{name:gbuppercase}[4]{\bibauthorfont%
  \gbcaselocalset%
  \ifuseprefix%
    {\usebibmacro{name:delim}{#3#1}%
     \usebibmacro{name:hook}{#3#1}%
     \ifdefvoid{#3}{}{%
       \ifcapital%
         {\mkbibnameprefix{\MakeLowercase{#3}}\isdot}%
         {\mkbibnameprefix{#3}\isdot}%
       \ifprefchar{}{\bibnamedelimc}}%
     \ifdefvoid{#2}{\mkbibnamefamily{\MakeCapital{#1}}}{\mkbibnamefamily{\MakeUppercase{#1}}}\isdot%
     \ifdefvoid{#2}{}{\revsdnamepunct\bibnamedelimd\mkbibnamegiven{\MakeUppercase{#2}}\isdot}%\MakeCapital
     \ifdefvoid{#4}{}{\addcomma\bibnamedelimd\mkbibnamesuffix{#4}\isdot}}%后缀前加逗号
    {\usebibmacro{name:delim}{#1}%
     \usebibmacro{name:hook}{#1}%
     \ifdefvoid{#2}{\mkbibnamefamily{\MakeCapital{#1}}}{\mkbibnamefamily{\MakeUppercase{#1}}}\isdot%
     \ifboolexpe{%
       test {\ifdefvoid{#2}}
       and
       test {\ifdefvoid{#3}}}
       {}{\revsdnamepunct}%
     \ifdefvoid{#2}{}{\bibnamedelimd\mkbibnamegiven{\MakeUppercase{#2}}\isdot}%\MakeCapital
     \ifdefvoid{#3}{}{\bibnamedelimd\mkbibnameprefix{#3}\isdot}%
     \ifdefvoid{#4}{}{\addcomma\bibnamedelimd\mkbibnamesuffix{#4}\isdot}}%%后缀前加逗号
}
text\nocite{*}
\citep{mikkonen2018,first,second}
\printbibliography

\end{document} 
```
![image](https://github.com/hushidong/biblatex-gb7714-2015/assets/45897471/941c8055-af08-4514-ac8e-e7846a9a0853)

三个文件均已下载：
![image](https://github.com/hushidong/biblatex-gb7714-2015/assets/45897471/22b3b6e0-b146-4a89-8be5-e2ce7f57e0fa)

**Comment by hushidong** [comment]
你下的版本不对啊，你仔细点，我3点多才更新的，你这个文件都是2点多的。

遇到问题自己找找原因，尝试过各种可能后再把问题描述出来。老是让别人指出低级错误，对你没有好处的，看起来你喜欢钻研，但很不仔细。包括那个key排序的问题，你检查一下key的内容，我之前都已经给你指出来了，你还是在那反复问。

抱歉，我本不应该这么说的，但的确要提醒一下你。而且我感觉你有点走入误区了，咱们使用latex是为了方便快捷，自动化，又能够给科研提供帮助，又能省心专注于内容，格式模板都定义好了，实在没有必要投入太多时间去改，除非你是要做模板。
说实话好看与否真的是非常次要的，我看你过于追求格式了，纠结一个空格或者排序对你论文有什么提升呢？这又不是导师非要你改的？

抱歉，我可能说多了，可能让你不舒服，如果你觉得不对就当没看见就行。

还是回到事情上来，你遇到问题，你首先要深入一点，仔细一点，在充分分析后，然后再拿出来讨论，不要拿一个明显简单错误的问题问。

**Reply by earthmoon** [reply]
> 你下的版本不对啊，你仔细点，我3点多才更新的，你这个文件都是2点多的。
> 
> 遇到问题自己找找原因，尝试过各种可能后再把问题描述出来。老是让别人指出低级错误，对你没有好处的，看起来你喜欢钻研，但很不仔细。包括那个key排序的问题，你检查一下key的内容，我之前都已经给你指出来了，你还是在那反复问。
> 
> 抱歉，我本不应该这么说的，但的确要提醒一下你。而且我感觉你有点走入误区了，咱们使用latex是为了方便快捷，自动化，又能够给科研提供帮助，又能省心专注于内容，格式模板都定义好了，实在没有必要投入太多时间去改，除非你是要做模板。 说实话好看与否真的是非常次要的，我看你过于追求格式了，纠结一个空格或者排序对你论文有什么提升呢？这又不是导师非要你改的？
> 
> 抱歉，我可能说多了，可能让你不舒服，如果你觉得不对就当没看见就行。
> 
> 还是回到事情上来，你遇到问题，你首先要深入一点，仔细一点，在充分分析后，然后再拿出来讨论，不要拿一个明显简单错误的问题问。

没有不舒服的，我喜欢听真实的意见，好听不好听一点都不重要，能反映真实问题的都是极其好的，所以我非常感谢您。因为这里一看，所有的修改时间都在[3天前](https://github.com/hushidong/biblatex-gb7714-2015)以上。
![image](https://github.com/hushidong/biblatex-gb7714-2015/assets/45897471/e3d64e2b-2a66-4957-b32a-92263f8acd2a)

我不知道下载哪一个。所以我点击标题上的最新更新通知：
![image](https://github.com/hushidong/biblatex-gb7714-2015/assets/45897471/7d5f277e-bbf3-40e4-88a3-2e6a990211fe)

然后再点击
![image](https://github.com/hushidong/biblatex-gb7714-2015/assets/45897471/1001f9f5-6d61-41bd-ba1d-927f554b1822)

再点击下载
![image](https://github.com/hushidong/biblatex-gb7714-2015/assets/45897471/dc8d7c8b-9579-49c9-9860-7872f4ff75db)


结果就是老版本。我到现在也不知道怎么下载新版本，应该是我对github的熟悉程度很不够造成的。

**Comment by hushidong** [comment]
你可以看看别人的提问：

https://github.com/hushidong/biblatex-gb7714-2015/issues/166

https://github.com/hushidong/biblatex-gb7714-2015/issues/113

有充分的调研，充分的分析，合理的建议，这是我们作为维护者更希望看到的。

**Reply by earthmoon** [reply]
> 你可以看看别人的提问：
> 
> #166
> 
> #113
> 
> 有充分的调研，充分的分析，合理的建议，这是我们作为维护者更希望看到的。

是的，他们做的很充分。我主要只是考虑，如果没有空格，他们连两个分开的字都不算。因为在中文中，两个字之间是有间距的。在英文中，两个单词之间是有间距的。也只是一种感觉而已。
